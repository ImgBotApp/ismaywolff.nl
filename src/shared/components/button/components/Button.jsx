import styled from 'styled-components'
import { bool } from 'prop-types'

const Button = styled.button`
  background: var(--color-gray);
  border: 1px solid var(--color-gray-dark);
  border-radius: 0;
  display: ${props => (props.fill ? 'block' : 'unset')};
  padding: var(--size-smaller);
  width: ${props => (props.fill ? '100%' : 'unset')};
`

Button.propTypes = {
  fill: bool
}

Button.defaultProps = {
  fill: false
}

export default Button
