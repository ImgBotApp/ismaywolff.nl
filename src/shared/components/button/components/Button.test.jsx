/**
 * @jest-environment node
 */

import 'jest-styled-components'
import React from 'react'
import { shallow } from 'enzyme'
import Button from './Button'

describe('<Button />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Button to="/" />)
    expect(wrapper).toMatchStyledComponentsSnapshot()
  })

  it('accepts a dark bool', () => {
    const wrapper = shallow(<Button to="/" dark />)
    expect(wrapper).toMatchStyledComponentsSnapshot()
  })
})
