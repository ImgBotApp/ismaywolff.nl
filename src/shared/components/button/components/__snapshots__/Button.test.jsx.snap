// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> accepts a dark bool 1`] = `
.xnCxw {
  background: var(--color-transparent-black);
  font-size: var(--size-small);
  padding: var(--size-smaller) var(--size-small);
  white-space: nowrap;
}

<span
  className="sc-bdVaJa xnCxw"
  to="/"
/>
`;

exports[`<Button /> renders correctly 1`] = `
.kQRvIH {
  background: unset;
  font-size: var(--size-small);
  padding: var(--size-smaller) var(--size-small);
  white-space: nowrap;
}

<span
  className="sc-bdVaJa kQRvIH"
  to="/"
/>
`;
